commons-oauth/src/main/java/eu/europeana/api/commons/oauth2/service/impl/EuropeanaClientDetailsService.java

```diff
@@ -18,61 +26,85 @@
-      	  throw new OAuth2Exception("Invocation of api key service failed. Cannot validate ApiKey : " + key, e);
-        }
+    public void setAuthHandler(AuthenticationHandler authHandler) {
+      this.authHandler = authHandler;
+    }
 
-	//validation is never null after the invokation of validateApiKyy
-	if (validation != null && validation.isValidApiKey()) {
-	    // valid api key
-	    return new ClientDetailsAdapter(key);
-	} 
-	
-	if (validation == null || HttpStatus.SC_INTERNAL_SERVER_ERROR == validation.getHttpStatus()) {
-	    // apikey service is faulty
-	    String message = (validation == null) ? "unknown" : validation.getErrorMessage(); 
-	    throw new OAuth2Exception("Invocation of api key service failed. Cannot validate ApiKey : " + key
-		    + ". Reason:" + message);
-	} else {
-	    // invalid apikey
-	    throw new ClientRegistrationException("Invalid API key: " + key + " Reason: " + validation.getHttpStatus()
-		    + " : " + validation.getErrorMessage());
-	}        	
+  @Override
+  /* Loads ClientDetails object belongs to an apiKey
+   */
+  public ClientDetails loadClientByClientId(String key) throws OAuth2Exception, ClientRegistrationException {
+    //allow disabling apikey validation for read access
+    if (StringUtils.isBlank(getApiKeyServiceUrl())) {
+      return null;
     }
-    
-    public String getApiKeyServiceUrl() {
-        return apiKeyServiceUrl;
+    KeyValidationResult result;
+    try {
+       result = validateApiKeyKeycloakClient(key);
+    } catch (ApiKeyValidationException | RuntimeException e) {
+      //service not accessible (e.g. IO Exception wrapped by the client, or other runtime exception )
+      throw new OAuth2Exception(
+          "Invocation of api key service failed. Cannot validate ApiKey : " + key, e);
+    }
+    //If the key validation result contains the error message return the error
+    if (result != null && result.getValidationError() !=null) {
+      throw new EuropeanaClientRegistrationException(null, result);
     }
+    // valid api key
+    return new ClientDetailsAdapter(key);
+  }
 
-    public void setApiKeyServiceUrl(String apiKeyServiceUrl) {
-        this.apiKeyServiceUrl = apiKeyServiceUrl;
+
+  /**
+   * Method calls the validation endpoint of keycloak to validate the input apikey returns
+   * no-content on success and  error response object upon failure.
+   *
+   * @return either null or result containing keycloak error messages
+   */
+    public KeyValidationResult validateApiKeyKeycloakClient(String apikey)
+        throws ApiKeyValidationException {
+
+      HttpConnection httpConnection = new HttpConnection();
+      try {
+        HttpResponseHandler response = httpConnection.post(VALIDATION_PARAMS.formatted(apiKeyServiceUrl,apikey),null,
+            MediaType.APPLICATION_JSON_VALUE, authHandler);
+
+        if(response==null || response.getStatus() == HttpStatus.SC_INTERNAL_SERVER_ERROR){
```

Comment:
It would be nice if we were consistent with our code formatting.  After `if` add a space , space in between `'=='`